api/recommendations.go

```diff
@@ -0,0 +1,101 @@
+//
+// Author:: Darren Murray (<darren.murray@lacework.net>)
+// Copyright:: Copyright 2022, Lacework Inc.
+// License:: Apache License, Version 2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package api
+
+import "fmt"
+
+// RecommendationsService is a service that interacts with the Recommendations
+// endpoints from the Lacework Server
+type RecommendationsService struct {
+	client *Client
+}
+
+type RecommendationType string
+
+const (
+	AwsRecommendation   RecommendationType = "aws"
+	AzureRecommendation RecommendationType = "azure"
+	GcpRecommendation   RecommendationType = "gcp"
+)
+
+func (svc *RecommendationsService) AwsList() ([]Recommendation, error) {
+	return svc.list(AwsRecommendation)
+}
+
+func (svc *RecommendationsService) AzureList() ([]Recommendation, error) {
+	return svc.list(AzureRecommendation)
+}
+
+func (svc *RecommendationsService) GcpList() ([]Recommendation, error) {
```

Comment:
There are two operation `List()` and `Patch()` for three clouds `Aws`, `Gcp`, and `Azure`. I think it makes
more sense to nest these as separate services inside Recommendations.
```
apiClient.Recommendations.Aws.List()
apiClient.Recommendations.Aws.Patch()
apiClient.Recommendations.Gcp.List()
apiClient.Recommendations.Gcp.Patch()
apiClient.Recommendations.Azure.List()
apiClient.Recommendations.Azure.Patch()